solver/src/main/java/org/chocosolver/solver/constraints/set/PropUnionVar.java

```diff
@@ -68,14 +67,36 @@ public PropUnionVar(SetVar union, int vOffset, SetVar indices, int iOffset, SetV
     //***********************************************************************************
     @Override
     public void propagate(int evtmask) throws ContradictionException {
-        SetVar union = vars[k];
         SetVar indices = vars[k + 1];
+        if (firstProp) {
+            firstProp = false;
+            model.getEnvironment().save(() -> firstProp = true);
```

Comment:
For my knowledge is there a big difference between this and using an IStateBool ?

Reply:
Ask yourself :)
The main interest of `IOperation` is that it is GC on backtrack where `IStateBool` is not.
Otherwise, it is comparable